gdal/frmts/rdb/rdbdataset.cpp

```diff
@@ -0,0 +1,784 @@
+    {
+        riegl::rdb::pointcloud::PointAttribute oAttribute =
+            oPointAttribute.get(osAttributeName);
+
+        if(osAttributeName == oPointAttribute.primaryAttributeName())
+        {
+            continue;
+        }
+        if(oAttribute.length == 1)
+        {
+            SetBandInternal(this, osAttributeName, oAttribute,
+                            oAttribute.dataType(), nNumberOfLevels - 1,
+                            nNumberOfLevels, nBandIndex);
+        }
+        else
+        {
+            for(uint32_t i = 0; i < oAttribute.length; i++)
+            {
+                std::ostringstream oOss;
+                oOss << osAttributeName << '[' << i << ']';
+                SetBandInternal(this, oOss.str(), oAttribute,
+                                oAttribute.dataType(), nNumberOfLevels - 1,
+                                nNumberOfLevels, nBandIndex);
+            }
+        }
+    }
+}
+
+GDALDataset *RDBDataset::Open(GDALOpenInfo *poOpenInfo)
+{
+    if(!Identify(poOpenInfo))
+    {
+        return nullptr;
+    }
+
+    if(poOpenInfo->eAccess == GA_Update)
+    {
+        CPLError(CE_Failure, CPLE_NotSupported,
+                 "The RDB driver does not support update access to existing "
+                 "datasets.");
+        return nullptr;
+    }
+
+    if(poOpenInfo->fpL == nullptr)
+    {
+        return nullptr;
+    }
+    try
+    {
+        RDBDataset *poDS = new RDBDataset(poOpenInfo);
+        // std::swap(poDS->fp, poOpenInfo->fpL);
+        // Initialize any PAM information.
+        poDS->SetDescription(poOpenInfo->pszFilename);
+        poDS->TryLoadXML();
+
+        return poDS;
+    }
+    catch(const riegl::rdb::Error &oException)
+    {
+        CPLError(CE_Failure, CPLE_OpenFailed, "RDB error: %s, %s",
+                 oException.what(), oException.details());
+        return nullptr;
+    }
+
+    catch(const std::exception &oException)
+    {
+        CPLError(CE_Failure, CPLE_OpenFailed, "Error: %s", oException.what());
+        return nullptr;
+    }
+    catch(...)
+    {
+        CPLError(CE_Failure, CPLE_OpenFailed, "Unknown error in Open.");
+        return nullptr;
+    }
+
+    return nullptr;
+}
+int RDBDataset::Identify(GDALOpenInfo *poOpenInfo)
+{
+    const char *psHeader = reinterpret_cast<char *>(poOpenInfo->pabyHeader);
+    if(poOpenInfo->nHeaderBytes < 32)
+    {
+        return FALSE;
+    }
+
+    constexpr int kSizeOfRDBHeaderIdentifier = 32;
+    constexpr char szRDBHeaderIdentifier[kSizeOfRDBHeaderIdentifier] =
+        "RIEGL LMS RDB 2 POINTCLOUD FILE";
+
+    if(strncmp(psHeader, szRDBHeaderIdentifier, kSizeOfRDBHeaderIdentifier))
+    {
+        // A more comprehensive test could be done by the library.
+        // Should file -> library incompatibilities handled in Identify or
+        // in the Open function?
+        return FALSE;
+    }
+    return TRUE;
+}
+
+CPLErr RDBDataset::GetGeoTransform(double *padfTransform)
+{
+    padfTransform[0] = dfXMin;
```

Comment:
yes, since the Webmercator coordinates start in southwest of the earth.

Reply:
> since the Webmercator coordinates start in southwest of the earth.

not sure that it answers my question, I just wanted to share attention that in most (like 99%) raster formats line 0 of the image is the one that is displayed at the top of your screen, that is the line with the max georeferenced Y value, hence padfTransform[3] is generally maxY, and padfTransform[5] a negative value. But I assume you've verified by gdal_translate'ing to GeoTIFF and displaying in QGIS or whatever that you get a correctly positionned raster.

Reply:
I see. You are correct. I used gdal_translate and dumpoverviews in order to create tif files from the .mpx file and checked if the data fits to a map.